resourcemanager/commonids/composite_resource_id.go

```diff
@@ -0,0 +1,83 @@
+package commonids
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+// CompositeResourceID is a struct representing the Resource ID for a Composite Resource Id
+type CompositeResourceID[T1 resourceids.ResourceId, T2 resourceids.ResourceId] struct {
+	First  T1
+	Second T2
```

Comment:
_minor_ but could we document these fields so that it's clearer what these are?

```suggestion
	// First specifies the first component of this Resource ID.
	// This is in the format `{first}|{second}`.
	First  T1
	
	// Second specifies the second component of this Resource ID
	// This is in the format `{first}|{second}`.
	Second T2
```